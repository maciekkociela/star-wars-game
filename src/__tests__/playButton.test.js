import React from "react";
import PlayButton from "../components/playButton";
import renderer from "react-test-renderer";

it("renders correctly", () => {
    const tree = renderer
        .create(<PlayButton loading={1} message="Test" />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders as a progress spinner when loading is 1", () => {
    const tree = renderer
        .create(<PlayButton loading={1} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders as a button when loading is 0", () => {
    const tree = renderer
      .create(<PlayButton loading={0} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
});
