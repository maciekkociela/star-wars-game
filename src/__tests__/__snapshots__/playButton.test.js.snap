// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as a button when loading is 0 1`] = `
<div
  style={
    Object {
      "margin": 10,
    }
  }
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Play
    </span>
  </button>
  <p>
    
  </p>
</div>
`;

exports[`renders as a progress spinner when loading is 0 1`] = `
<div
  style={
    Object {
      "margin": 10,
    }
  }
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Play
    </span>
  </button>
  <p>
    
  </p>
</div>
`;

exports[`renders as a progress spinner when loading is 1 1`] = `
<div
  style={
    Object {
      "margin": 10,
    }
  }
>
  <div
    className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
  <p>
    
  </p>
</div>
`;

exports[`renders as a progress spinner when loading us 0 1`] = `
<div
  style={
    Object {
      "margin": 10,
    }
  }
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Play
    </span>
  </button>
  <p>
    
  </p>
</div>
`;

exports[`renders correctly 1`] = `
<div
  style={
    Object {
      "margin": 10,
    }
  }
>
  <div
    className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
  <p>
    Test
  </p>
</div>
`;
